signer/cosigner_nonce_cache.go

```diff
@@ -341,21 +323,27 @@ CheckNoncesLoop:
 	return nil, fmt.Errorf("no nonces found involving cosigners %+v", cosignerInts)
 }
 
-func (cnc *CosignerNonceCache) PruneNonces() {
+func (cnc *CosignerNonceCache) PruneNonces() int {
 	cnc.cache.mu.Lock()
 	defer cnc.cache.mu.Unlock()
-	for u, cn := range cnc.cache.cache {
-		if time.Now().After(cn.Expiration) {
-			fmt.Printf("Pruning nonce %s\n", u)
-			delete(cnc.cache.cache, u)
+	nonExpiredIndex := len(cnc.cache.cache) - 1
+	for i := len(cnc.cache.cache) - 1; i >= 0; i-- {
+		if time.Now().Before(cnc.cache.cache[i].Expiration) {
```

Comment:
If the whole cache expires, it doesn't look like anything would be pruned

Reply:
good catch, fixed and added test